fix spelling of advertise (shows up in the debug log)

jloughry opened the pull request:
Minor spelling fix: the message `AdvertizeLocal: advertizing address 168.103.195.250:8333` in the debug log was misspelt.

jloughry committed:
fix spelling of advertise in src and doc

laanwj commented:
Isn't that just US versus UK spelling?

jloughry commented:
I've lived in US and UK and both countries spell it 'advertise'. It's just one of those weird edge cases in English.

References:

1. http://grammarist.com/spelling/advertise-vs-advertize/
2. The Oxford Style Guide ('Fowler') prefers '-ise' in this particular case.

paveljanik commented:
Can you please fix them all in the source when you are at it?

jloughry commented:
Fixed everywhere in src and doc.

paveljanik commented:
Please check by `git grep advertiz`.

jloughry commented:
You're right. The remaining ones have been caught and fixed now. You taught me a new tool---thanks!

paveljanik commented:
@jloughry great! Now can you please squash two commits into one? The same tool, different command line arguments :-)

jloughry commented:
I squashed all my changes into the 37767fd commit, but the pull request has 4 commits in it now: the two original ones that were squashed, plus something new: cb5844d, that appears to be an empty merge. I can't figure out how to clean up the pull request so it contains only one commit:

* 37767fd (This one contains all the changes: keep this one)
* 2ef3edf (first commit, squashed into 37767fd)
* 723551b (second commit, squashed into 37767fd)
* cb5944d (I don't know what this is; it seems to be an empty merge)

How do I edit the pull request to remove the unneeded commits? Or is this normal behaviour when squashing commits? Thanks for the guidance; I want to do it right.

achow101 commented:
@jloughry reset git so that HEAD is the commit you want it to be (37767fd) and then force push it to github

jloughry commented:
Done---there's only one commit in the pull request now. Thank you for showing me how to do it.

paveljanik commented:
ACK

laanwj commented:
@jloughry Thanks for the explanation, can't believe this was missed all this time but here we are.

Concept ACK.

MarcoFalke commented:
You can use *-ize as well: https://en.wiktionary.org/wiki/advertize. Still, I am ok with 37767fd if it get's backported, so we don't create unnecessary backport conflicts.

jloughry commented:
A question of procedure: should backport changes be added to this pull request, or made one or more new pull requests?

sipa commented:
You should typically only create pull requests against master; we'll backport changes to older releases as needed. In case a patch specifically applies to a backport only (like a change to 0.12 release notes) or is highly nontrivial to apply to older code (due to refactoring that may have taken place), you can create pull requests against specific branches.

jloughry commented:
Thanks for explaining it.

laanwj commented:
So what is it, can you use advertize or not?
If it is remotely valid, I'd prefer to keep it as it (as @marcofalke says this will create rebasing conflicts), if it looks stupid to native English speakers it should be changed.

wtogami commented:
It looks stupid to me, a native English speaker.  I might have used it a few times only as it seemed to be an awkward Bitcoin term and I was talking about the peer manager at the time.

MarcoFalke commented:
Backported as part of #7938. Removing label 'Needs backport'.
